src/main/java/dk/kb/util/other/StringListUtils.java

```diff
@@ -0,0 +1,281 @@
+package dk.kb.util.other;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class StringListUtils {
+    
+    
+    public static final Predicate<String> notNull = Predicate.not(Objects::isNull);
+    public static final Predicate<String> notEmptyString = Predicate.not(String::isEmpty);
+    public static final Predicate<String> isNotEmpty = notNull.and(notEmptyString);
+    
+    /**
+     * Turn an iterator into a stream
+     *
+     * @param iterator the iterator
+     * @param <T>      the the type of elements
+     * @return A stream of the iterator elements in the same order
+     */
+    public static <T> Stream<T> asStream(Iterator<T> iterator) {
+        return StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED),
+                false);
+    }
+    
+    /**
+     * Return the string or "" if null
+     *
+     * @param string the string
+     * @return same or "" if null
+     */
+    public static String notNull(String string) {
+        return Optional.ofNullable(string).orElse("");
+    }
+    
+    /**
+     * Return the first entry in the list or defaultValue if the list is empty
+     *
+     * @param list         the list
+     * @param defaultValue the value if the list is empty
+     * @return first value or default value if list is empty
+     */
+    public static String getFirst(List<String> list, String defaultValue) {
+        if (list == null || list.isEmpty()) {
+            return defaultValue;
+        } else {
+            return list.get(0);
+        }
+    }
+    
+    /**
+     * If param is not null return that. Otherwise return the value from props[propKey]
+     *
+     * @param props   the properties to default to
+     * @param param   the value to return if notNull
+     * @param propKey the key to props if param is null
+     * @return param or props[propKey]
+     */
+    public static String getParamOrProps(Properties props, String param, String propKey) {
```

Comment:
Could the argument order be `param`, `props`, `propKey` instead? That way it would match the `attemptedValue`, `fallbackValue`-pattern used elsewhere (e.g. in `getFirst`).

Reply:
Done